scripts/common.js

```diff
@@ -1,13 +1,320 @@
-export let commonOptions = {
-    stages: [
-        {duration: '1m', target: 5}, //5 virtual users over 1 minutes
-        {duration: '3m', target: 10}, //10 virtual users over 3 minutes
-        {duration: '10s', target: 0},
-    ]
+export let timeMapOptions = {
+    stages: [{
+            duration: '1m',
+            target: 5
+        }, //5 virtual users over 1 minutes
+        {
+            duration: '3m',
+            target: 10
+        }, //10 virtual users over 3 minutes
+        {
+            duration: '10s',
+            target: 0
+        },
+    ],
+    summaryTrendStats: ['avg', 'min', 'max', 'p(90)', 'p(95)'],
+    thresholds: {
+        'http_req_duration': ['max>=0'],
+        'http_req_receiving': ['max>=0'],
+        'http_req_sending': ['max>=0'],
+    }
 };
 
+export const destinations = (__ENV.DESTINATIONS || '50, 100, 150')
+    .split(',')
+    .map((curDestinations) => parseInt(curDestinations))
 
-export const countries = {
+const scenarios = destinations.reduce((accumulator, currentDestinations) => {
+    accumulator[`sending_${currentDestinations}_destinations`] = {
+        executor: 'constant-vus',
+        duration: '3m',
+        env: {SCENARIO_DESTINATIONS: currentDestinations.toString()},
+
+        vus: 1,
+        startTime: '10s',
+        gracefulStop: '10s',
+    }
+    return accumulator
+}, {})
+
+export const timeFilterOptions = {
+    scenarios: scenarios,
+    summaryTrendStats: ['avg', 'min', 'max', 'p(90)', 'p(95)'],
+
+    thresholds: {
+        // Intentionally empty. I'll define bogus thresholds (to generate the sub-metrics) below.
+    }
+}
+
+export function setThresholdsForScenarios(options) {
+    for (let key in options.scenarios) {
+        options.thresholds[`http_req_duration{scenario:${key}}`] = ['max>=0'];
+        options.thresholds[`http_req_receiving{scenario:${key}}`] = ['max>=0'];
+        options.thresholds[`http_req_sending{scenario:${key}}`] = ['max>=0'];
+    }
+};
+
+const countries = `GS,-54.283333,-36.5
```

Comment:
not a string, please. Only map: 'lv': {lat: 56.945614, lng: 24.120870},

Reply:
Done